src/containerManager.ts

```diff
@@ -31,40 +33,72 @@ export class ContainerManager {
     try {
       if (currentApp) {
         const containerName = currentApp.containerName;
-        if (!this.checkContainerExists(containerName)) {
-          this.statusBarManager.updateDevImageItem(DevImageStatus.stopped);
-          this.treeProvider.updateContainerLabel(DevImageStatus.stopped);
-          console.log(`Ledger: Container ${currentApp.containerName} not found, respawning it.`);
-          this.taskProvider.executeTaskByName("Update Container");
-        } else {
+
+        if (this.checkContainerExists(containerName)) {
           const command = `docker inspect -f "{{ .State.Status }}" ${containerName}`;
           const containerStatus = execSync(command).toString().trim();
+          console.log(`Ledger: Container ${containerName} status is ${containerStatus}`);
+
           if (containerStatus === "running") {
             this.statusBarManager.updateDevImageItem(DevImageStatus.running);
             this.treeProvider.updateContainerLabel(DevImageStatus.running);
-          } else if (containerStatus === "starting" || containerStatus === "restarting") {
+            const conf = vscode.workspace.getConfiguration("ledgerDevTools");
+            if (conf.get<boolean>("keepContainerTerminal") == false) {
+                  vscode.window.activeTerminal?.hide()
+            }
+            return true;
+          }
+          if (containerStatus === "starting" || containerStatus === "restarting") {
             this.statusBarManager.updateDevImageItem(DevImageStatus.syncing);
             this.treeProvider.updateContainerLabel(DevImageStatus.syncing);
-          } else {
-            console.log(`Ledger: Container ${currentApp.containerName} status is ${containerStatus}, respawning it.`);
-            this.statusBarManager.updateDevImageItem(DevImageStatus.stopped);
-            this.treeProvider.updateContainerLabel(DevImageStatus.stopped);
-            this.taskProvider.executeTaskByName("Update Container");
+            return true;
           }
         }
-      } else {
-        this.statusBarManager.updateDevImageItem(DevImageStatus.stopped);
-        this.treeProvider.updateContainerLabel(DevImageStatus.stopped);
       }
+      return false;
+
     } catch (error: any) {
       console.log(`Docker error : ${error.message}`);
-      this.statusBarManager.updateDevImageItem(DevImageStatus.stopped);
-      this.treeProvider.updateContainerLabel(DevImageStatus.stopped);
+      return false;
+    }
+  }
+
+  manageContainer(): void {
+    if (this.isContainerReady() == false) {
+      const currentApp = getSelectedApp();
+      if (currentApp) {
+        this.statusBarManager.updateDevImageItem(DevImageStatus.stopped);
+        this.treeProvider.updateContainerLabel(DevImageStatus.stopped);
+        console.log(`Ledger: Container ${currentApp.containerName} not found, respawning it.`);
+        this.taskProvider.executeTaskByName("Update Container");
+      }
+    }
+  }
+
+  checkRetries(): void {
```

Comment:
@cedelavergne-ledger maybe change the name to something a little bit more explicit like `checkUpdateRetries`

Reply:
Done!